src/charm.py

```diff
@@ -136,6 +137,11 @@ def _on_config_changed(self, _: typing.Any) -> None:
         """Handle the config-changed event."""
         self._reconcile()
 
+    def _on_upgrade_charm(self, _: typing.Any) -> None:
+        """Handle the upgrade-charm event."""
+        self.haproxy_service.install()
```

Comment:
Since config-changed is called after both, do you think would work to add this to the reconcile and remove both on_install/on_upgrade events?

Reply:
good idea, I moved it to the reconcile method